ReactApp1/reactapp1.client/src/components/Domain/Order/Stripe.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { CardElement, useStripe, useElements } from '@stripe/react-stripe-js';
+import { Form } from 'react-bootstrap';
+import axios from 'axios';
+import ScriptResources from "../../../assets/resources/strings.ts";
+import { Order } from "../../../components/Domain/Order/Orders.tsx";
+
+interface StripePaymentProps {
+    order: Order | undefined;
+    token: string | undefined;
+    paymentValue: number;
+    setPaymentValue: React.Dispatch<React.SetStateAction<number>>; 
+    handlePayPayment: () => void;
+}
+
+const StripePayment: React.FC<StripePaymentProps> = ({
+    order,
+    token,
+    paymentValue,
+    setPaymentValue,
+    handlePayPayment,
+}) => {
+
+    const stripe = useStripe();
+    const elements = useElements();
+
+    const cardOptions = {
+        hidePostalCode: true
+    };
+
+    const handleStripePayment = async () => {
+        if (order?.orderId) {
+            try {
+                if (!stripe || !elements) {
+                    alert(`Stripe initialization failed`);
+                    return;
+                }
+
+                const cardElement = elements.getElement(CardElement);
+                if (!cardElement) {
+                    alert(`Payment initialization failed`);
+                    return;
+                }
+
+                const { data: clientSecret } = await axios.post(
+                    `http://localhost:5114/api/payments/createIntent/${paymentValue}/${ScriptResources.Currency}`,
+                    {},
+                    { headers: { Authorization: `Bearer ${token}` } }
+                );
+
+                const { error, paymentIntent } = await stripe.confirmCardPayment(clientSecret.clientSecret, {
+                    payment_method: {
+                        card: cardElement,
+                    },
+                });
+
+                if (error) {
+                    alert(`Payment failed: ${error.message}`);
+                } else if (paymentIntent.status === 'succeeded') {
+                    alert("Payment successful");
+                    handlePayPayment();
+                }
+
+            } catch (error) {
+                console.error("Error in Stripe payment: ", error);
+                alert("Payment failed.");
```

Comment:
Šitą alertą gali ištrint ir geriau iš BE grąžinti custom exception. Kaip dabar padaryta prie Item ir Order. Yra globalus errorų handlinimas.
Jeigu kažkur dar pamatysi alertus kurie iššoka kai iš BE atsiranda erroras, gali ištrint.